code/game/machinery/station_map.dm

```diff
@@ -0,0 +1,321 @@
+	component_parts = 0
+	if(ticker && holomaps_initialized)
+		initialize()
+
+/obj/machinery/station_map/Destroy()
+	station_holomaps -= src
+	stopWatching()
+	..()
+
+/obj/machinery/station_map/crowbarDestroy(mob/user)
+	user.visible_message(	"[user] begins to pry out \the [src] from the wall.",
+							"You begin to pry out \the [src] from the wall...")
+	if(do_after(user, src, 40))
+		user.visible_message(	"[user] detaches \the [src] from the wall.",
+								"You detach \the [src] from the wall.")
+		playsound(get_turf(src), 'sound/items/Crowbar.ogg', 50, 1)
+		new /obj/item/mounted/frame/station_map(src.loc)
+
+		for(var/obj/I in src)
+			qdel(I)
+
+		new /obj/item/weapon/circuitboard/station_map(src.loc)
+		new /obj/item/stack/cable_coil(loc,5)
+		new /obj/item/stack/sheet/glass/glass(loc,1)
+
+		return 1
+	return -1
+
+/obj/machinery/station_map/initialize()
+	bogus = 0
+	original_zLevel = loc.z
+	if(!(HOLOMAP_EXTRA_STATIONMAP+"_[original_zLevel]" in extraMiniMaps))
+		bogus = 1
+		station_map = image('icons/480x480.dmi', "stationmap")
+		legend = image('icons/effects/64x64.dmi', "notfound")
+		legend.pixel_x = 7*WORLD_ICON_SIZE
+		legend.pixel_y = 7*WORLD_ICON_SIZE
+		update_icon()
+		return
+
+	station_map = image(extraMiniMaps[HOLOMAP_EXTRA_STATIONMAP+"_[original_zLevel]"])
+	small_station_map = image(extraMiniMaps[HOLOMAP_EXTRA_STATIONMAPSMALL_NORTH+"_[original_zLevel]"])
+	small_station_map.plane = LIGHTING_PLANE
+	small_station_map.layer = ABOVE_LIGHTING_LAYER
+	cursor = image('icons/holomap_markers.dmi', "you")
+	if(map.holomap_offset_x.len >= original_zLevel)
+		cursor.pixel_x = (x-6+map.holomap_offset_x[original_zLevel])*PIXEL_MULTIPLIER
+		cursor.pixel_y = (y-6+map.holomap_offset_y[original_zLevel])*PIXEL_MULTIPLIER
+	else
+		cursor.pixel_x = (x-6)*PIXEL_MULTIPLIER
+		cursor.pixel_y = (y-6)*PIXEL_MULTIPLIER
+	legend = image('icons/effects/64x64.dmi', "legend")
+	legend.pixel_x = 3*WORLD_ICON_SIZE
+	legend.pixel_y = 3*WORLD_ICON_SIZE
+	floor_markings = image('icons/turf/overlays.dmi', "station_map")
+	floor_markings.dir = dir
+	floor_markings.plane = ABOVE_TURF_PLANE
+	floor_markings.layer = DECAL_LAYER
+	station_map.overlays |= cursor
+	station_map.overlays |= legend
+	update_icon()
+
+/obj/machinery/station_map/attack_hand(var/mob/user)
+	if(watching_mob && (watching_mob != user))
+		to_chat(user, "<span class='warning'>Someone else is currently watching the holomap.</span>")
+		return
+
+	if(user.loc != loc)
+		to_chat(user, "<span class='warning'>You need to stand in front of \the [src].</span>")
+		return
+
+	if(isliving(user) && anchored && !(stat & (NOPOWER|BROKEN)))
+		if(user.hud_used && user.hud_used.holomap_obj)
+			station_map.loc = user.hud_used.holomap_obj
+			station_map.alpha = 0
+			animate(station_map, alpha = 255, time = 5, easing = LINEAR_EASING)
+			watching_mob = user
+			flick("station_map_activate", src)
+			watching_mob.client.images |= station_map
+			watching_mob.callOnFace |= "\ref[src]"
```

Comment:
This line doesn't do anything, setting a nonexistant key on a list always creates the key.